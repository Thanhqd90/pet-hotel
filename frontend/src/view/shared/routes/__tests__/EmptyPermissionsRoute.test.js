import { mount } from 'enzyme';
import React, { Component } from 'react';
import {
  BrowserRouter,
  Link,
  Route,
  Switch,
} from 'react-router-dom';
import EmptyPermissionsRoute from 'view/shared/routes/EmptyPermissionsRoute';

function mountPage(currentUser) {
  const EmptyPermissionsPage = class EmptyPermissionsPage extends Component {
    render() {
      return (
        <div data-test="EmptyPermissionsPage">
          Empty Permissions Page
        </div>
      );
    }
  };

  return mount(
    <BrowserRouter>
      <div>
        <Link to="/auth/empty-permissions">
          Go to Empty Permissions
        </Link>

        <Switch>
          <Route
            exact
            path="/auth/signin"
            render={() => (
              <div data-test="SignInPage">Sign in</div>
            )}
          />

          <Route
            exact
            path="/"
            render={() => (
              <div data-test="PrivatePage">Private</div>
            )}
          />

          <EmptyPermissionsRoute
            path="/auth/empty-permissions"
            currentUser={currentUser}
            component={EmptyPermissionsPage}
          />
        </Switch>
      </div>
    </BrowserRouter>,
  );
}

describe('view/shared/routes/EmptyPermissionsRoute', () => {
  describe('Authenticated with permissions', () => {
    let component;
    beforeAll(() => {
      component = mountPage({
        id: 1,
        roles: ['viewer'],
        emailVerified: true,
      });

      component
        .find('Link[to="/auth/empty-permissions"]')
        .simulate('click', { button: 0 });
    });

    test('redirect to private', () => {
      expect(component.html()).toMatchSnapshot();

      expect(
        component
          .find(`[data-test="PrivatePage"]`)
          .exists(),
      ).toBeTruthy();
    });
  });

  describe('Authenticated with NO permissions', () => {
    let component;
    beforeAll(() => {
      component = mountPage({
        id: 1,
        roles: [],
        emailVerified: false,
      });

      component
        .find('Link[to="/auth/empty-permissions"]')
        .simulate('click', { button: 0 });
    });

    test('opens the page', () => {
      expect(component.html()).toMatchSnapshot();

      expect(
        component
          .find(`[data-test="EmptyPermissionsPage"]`)
          .exists(),
      ).toBeTruthy();
    });
  });

  describe('Not authenticated', () => {
    let component;

    beforeAll(() => {
      component = mountPage(null);

      component
        .find('Link[to="/auth/empty-permissions"]')
        .simulate('click', { button: 0 });
    });

    test('redirect to signin', () => {
      expect(component.html()).toMatchSnapshot();

      expect(
        component.find(`[data-test="SignInPage"]`).exists(),
      ).toBeTruthy();
    });
  });
});
