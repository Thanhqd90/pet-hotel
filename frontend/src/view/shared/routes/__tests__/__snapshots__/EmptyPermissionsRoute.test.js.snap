// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view/shared/routes/EmptyPermissionsRoute Authenticated with NO permissions opens the page 1`] = `"<div><a href=\\"/auth/empty-permissions\\">Go to Empty Permissions</a><div data-test=\\"EmptyPermissionsPage\\">Empty Permissions Page</div></div>"`;

exports[`view/shared/routes/EmptyPermissionsRoute Authenticated with permissions redirect to private 1`] = `"<div><a href=\\"/auth/empty-permissions\\">Go to Empty Permissions</a><div data-test=\\"PrivatePage\\">Private</div></div>"`;

exports[`view/shared/routes/EmptyPermissionsRoute Not authenticated redirect to signin 1`] = `"<div><a href=\\"/auth/empty-permissions\\">Go to Empty Permissions</a><div data-test=\\"SignInPage\\">Sign in</div></div>"`;
